preprocessing/data_science_help_functions.py

```diff
@@ -189,3 +197,178 @@ def detect_id_target_problem(dataframes_dict: dict, threshold: float = 0.1) -> T
     print(f"The possible possible_target are:\n {possible_target}")
     print(f"The type of the problem that should be solved:\n {possible_problems}")
     return possible_ids, possible_target, possible_problems
+
+
+def adversarial_validation(dataframe_dict: dict,
+                           ignore_columns: list = [],
+                           max_dataframe_length: int = 100000,
+                           threshold: float = 0.7) -> float:
+    """ Make adversarial validation checking
+
+    Training a probabilistic classifier to distinguish train/test examples.
+    See more info here: http://fastml.com/adversarial-validation-part-one/
+    This function checks whether test and train data coming from the same data distribution.
+
+    :param dict dataframes_dict: A dictionary that contains pandas dataframes e.g. dataframes_dictionary ={
+                'train': train_dataframe, 'test': test_dataframe}
+    :param int max_dataframe_length: Max length of dataframe to be considered - make adversarial validation faster
+    :param list ignore_columns: List of column to ignore (ID, target, etc...)
+    :param float threshold: A value larger than 0 and less than 1. If the result of calculation is greater than threshold - there is sugnificant difference between train and test data
+
+    :return:
+            - adversarial_validation_result: Adversarial validation score.
+    """
+
+    # Check if it only one dataframe provided
+    if len(dataframe_dict) != 2:
+        # do nothing and return the original data
+        print("Can't apply adversarial_validation because count of dataframes is not equal to 2")
+        return
```

Comment:
Thanks a lot for your great refactoring. You have a "return" statement here. I think you forgot removing it. I tried to test your code in one of the flows, but it did not work. I would suggest testing your code in all flows and check if it works without any problem. Even better, I would recommend strongly to write unit tests for the functions that you created.